src/app/hooks/useClickTracker/index.jsx

```diff
@@ -0,0 +1,72 @@
+import { useEffect, useRef, useContext, useCallback } from 'react';
+import { sendEventBeacon } from '#containers/ATIAnalytics/beacon/index';
+import { getComponentInfo } from '#app/lib/analyticsUtils/index';
+import { ServiceContext } from '#contexts/ServiceContext';
+import { RequestContext } from '#contexts/RequestContext';
+import { buildATIClickParams } from '#containers/ATIAnalytics/params';
+import { isValidClick } from './helpers';
+
+const useClickTracker = ({ pageData, componentName } = {}) => {
+  let eventTrackingProps;
+
+  const requestContext = useContext(RequestContext);
+  const serviceContext = useContext(ServiceContext);
+  const { service } = serviceContext;
+  const clickRef = useRef(null);
+
+  try {
+    eventTrackingProps = buildATIClickParams(
+      pageData,
+      requestContext,
+      serviceContext,
+    );
+  } catch (e) {
+    eventTrackingProps = null;
```

Comment:
As pointed out by @andrewscfc in my PR we should probably log something to the browser console until we have client side logging in place.

This is what I did https://github.com/bbc/simorgh/pull/9071/files#diff-7847911c4a756dfc4dce2205ea165af35dbdbb6c296083a0a02862757737cb92R41-R43

and then I ensured we have logging in the unit tests https://github.com/bbc/simorgh/pull/9071/files#diff-81136bbf60a2de9e097e285911100a0850d8b5dc7985bee4376851cf43e47da8R211-R215